Tasks/Xcode/xcode.ts

```diff
@@ -369,6 +369,31 @@ async function run() {
                         if (exportTeamId) {
                             tl.tool(plist).arg(['-c', 'Add teamID string ' + exportTeamId, exportOptionsPlist]).execSync();
                         }
+
+                        if (xcodeVersion >= 9 && !automaticSigningWithXcode && exportOptions === 'auto') {
+                            // Xcode 9 manual signing, set code sign style = manual
+                            tl.tool(plist).arg(['-c', 'Add signingStyle string ' + 'manual', exportOptionsPlist]).execSync();
+
+                            // add provisioning profiles to the exportOptions plist
+                            // find bundle Id from Info.plist and prov profile name from the embedded profile in each .app package
+                            tl.tool(plist).arg(['-c', 'Add provisioningProfiles dict', exportOptionsPlist]).execSync();
+
+                            for (let i = 0; i < embeddedProvProfiles.length; i++) {
```

Comment:
Loop over all the embedded.mobileprovision and the matching Info.plist to get the right profile name and bundle identifiers when multiple apps are in the .xcarchive.